### DISCUSSION TITLE
HELP ! failed to open file in client server.

### DISCUSSION THREAD

**Discussion by mocakbarmaulana** [ONLYOFFICE/DocumentServer#1539] [Q&A]
why i'am failed to open file in client server?  i'am following step in basic concepts document server.  but i dont understand how to get the key, i just copy and replace the key form documentation. the page client server is successfully load with onlyoffice docx, but the content isn't showing..?

**Comment by ShockwaveNN** [comment]
Key should be unique for each document
This may be just a random string

**Reply by mocakbarmaulana** [reply]
i change the key with random string, but the page is error occurs. and if i change again with key in basic concept tutorial. it's work but the content isn't showing..

**Reply by ShockwaveNN** [reply]
How is the error looked?
That is the message in DocumentServer logs located at `/var/log/onlyoffice/documentserver`

Also you can setup DocumentServer via docker and check integrated example - you can compare params of opening document and see that you doing wrong for your custom integration (am I right that you've using some your own integration?)

**Reply by mocakbarmaulana** [reply]
iam using in linux server. and iam just try to test opening file using simple html and js. but its not working. 
iam sorry but what a name file for looking the error log?

**Reply by ShockwaveNN** [reply]
> iam sorry but what a name file for looking the error log?

Any error
If documentserver is setup properly there should be no error at all, in all script files

**Reply by mocakbarmaulana** [reply]
this is no error right?
![ss-log](https://user-images.githubusercontent.com/78366474/144806026-406c397c-84ea-449b-ad89-0b4c28d4c47c.png)

**Reply by ShockwaveNN** [reply]
This is incorrect path, read again 

`/var/log/onlyoffice/documentserver`, not `/etc/onlyoffice/documentserver`

**Reply by mocakbarmaulana** [reply]
021/12/06 06:59:08 [error] 21269#21269: *1192 upstream prematurely closed connection while reading response header from upstream, client: 50.50.50.100, server: , request: "POST /6.4.2-6/doc/Khirz6zTPdfd7/c/84>
2021/12/06 06:59:08 [error] 21269#21269: *1194 upstream prematurely closed connection while reading response header from upstream, client: 50.50.50.100, server: , request: "POST /6.4.2-6/doc/Khirz6zTPdfd7/c/84>
2021/12/06 06:59:11 [error] 21269#21269: *1196 connect() failed (111: Connection refused) while connecting to upstream, client: 50.50.50.100, server: , request: "GET /6.4.2-6/doc/Khirz6zTPdfd7/c/info?t=1638773>
2021/12/06 06:59:11 [error] 21269#21269: *1196 connect() failed (111: Connection refused) while connecting to upstream, client: 50.50.50.100, server: , request: "GET /6.4.2-6/doc/Khirz6zTPdfd7/c/info?t=1638773>
2021/12/06 06:59:53 [error] 21269#21269: *1285 open() "/var/www/onlyoffice/documentserver/sdkjs/common/libfont/wasm/engine.js" failed (2: No such file or directory), client: 50.50.50.100, server: , request: "G>
2021/12/06 06:59:53 [error] 21269#21269: *1286 open() "/var/www/onlyoffice/documentserver/sdkjs/common/libfont/wasm/file.js" failed (2: No such file or directory), client: 50.50.50.100, server: , request: "GET>
2021/12/06 06:59:53 [error] 21269#21269: *1287 open() "/var/www/onlyoffice/documentserver/sdkjs/common/libfont/wasm/manager.js" failed (2: No such file or directory), client: 50.50.50.100, server: , request: ">
2021/12/06 06:59:53 [error] 21269#21269: *1288 open() "/var/www/onlyoffice/documentserver/sdkjs/common/libfont/wasm/file.js" failed (2: No such file or directory), client: 50.50.50.100, server: , request: "GET>
2021/12/06 06:59:53 [error] 21269#21269: *1289 open() "/var/www/onlyoffice/documentserver/sdkjs/common/libfont/wasm/engine.js" failed (2: No such file or directory), client: 50.50.50.100, server: , request: "G>
2021/12/06 06:59:53 [error] 21269#21269: *1290 open() "/var/www/onlyoffice/documentserver/sdkjs/common/libfont/wasm/manager.js" failed (2: No such file or directory), client: 50.50.50.100, server: , request: ">
2021/12/06 06:59:53 [error] 21269#21269: *1291 open() "/var/www/onlyoffice/documentserver/sdkjs/common/libfont/wasm/engine.js" failed (2: No such file or directory), client: 50.50.50.100, server: , request: "G>
2021/12/06 06:59:53 [error] 21269#21269: *1292 open() "/var/www/onlyoffice/documentserver/sdkjs/common/libfont/wasm/file.js" failed (2: No such file or directory), client: 50.50.50.100, server: , request: "GET>
2021/12/06 06:59:53 [error] 21269#21269: *1293 open() "/var/www/onlyoffice/documentserver/sdkjs/common/libfont/wasm/manager.js" failed (2: No such file or directory), client: 50.50.50.100, server: , request: ">

this is error what iam get in nginx.error.log.

**Reply by ShockwaveNN** [reply]
Sorry, my bad

I've sadi that

> If documentserver is setup properly there should be no error at all, in all script files

But in nginx.log can be errors related to `wasm`, but not other

Like:
```
021/12/06 06:59:08 [error] 21269#21269: *1192 upstream prematurely closed connection while reading response header from upstream, client: 50.50.50.100, server: , request: "POST /6.4.2-6/doc/Khirz6zTPdfd7/c/84>
```

Is there any other errors in other files?

**Reply by mocakbarmaulana** [reply]
have 3 other folder, converter, docservice, metrics.

**Reply by ShockwaveNN** [reply]
Any errors in log files in those folders?

**Reply by mocakbarmaulana** [reply]
how to get key unique if i upload file docx in url/example ?

**Reply by ShockwaveNN** [reply]
> how to get key unique if i upload file docx in url/example ?

If you mean integrated example -  keys unique by design

For example for nodejs example https://github.com/ONLYOFFICE/document-server-integration/blob/3a11a42f9ea8a59d860e29213eef1126320bed9d/web/documentserver-example/nodejs/helpers/documentService.js#L94

**Reply by mocakbarmaulana** [reply]
iam successfully load content file docx, but if i create new file using client side server. the new file isnt showing in simple doc management system. but if i create file or upload file using simple doc management system the file i created is showing in management system. how can i save document i created in client side and displaying in simple doc management system?

**Reply by ShockwaveNN** [reply]
>  the new file isnt showing in simple doc management system

Sorry I do not understand, do you mean "File -> Create New" button in DocumentServer

If not I think I'll need a video. because your description is very confusing for me

**Reply by mocakbarmaulana** [reply]
how can i save document with the callback?  i don't know how to make callback? sorry for bad english.. 
![image](https://user-images.githubusercontent.com/78366474/145159542-fc7fb0b4-d5ca-463c-b116-3d60fd903af1.png)

**Reply by ShockwaveNN** [reply]
You can open code of any our example (preferrable nodejs one) and see how callbacks configured

**Reply by mocakbarmaulana** [reply]
is there any with vanila javascript?

**Reply by ShockwaveNN** [reply]
Yep
https://github.com/ONLYOFFICE/document-server-integration/tree/master/web/documentserver-example/js

But it may be not feature reach, most advanced is `nodejs`
But callback logick should be in any example

**Reply by mocakbarmaulana** [reply]
okay... i'll try it.

**Reply by mocakbarmaulana** [reply]
https://api.onlyoffice.com/portals/basic is document server inluced with portal? and if not included how to using portal with document server?

**Reply by ShockwaveNN** [reply]
> is document server inluced with portal? and if not included how to using portal with document server?

What do you mean by `portal`? ONLYOFFICE CommunityServer or something other?

**Reply by mocakbarmaulana** [reply]
yup, it's onlyoffice document server and onlyoffice community server is different? how can i install onlyoffice coumminityserver ?

**Reply by ShockwaveNN** [reply]
Those are completly different product

CommunityServer is DMS, CMS, Project management and a lot of other features

> how can i install onlyoffice coumminityserver

Please, use google, it's not so hard to find those information

https://helpcenter.onlyoffice.com/installation/groups-install-ubuntu.aspx
Or you can read our [official site](https://onlyoffice.com)

**Reply by mocakbarmaulana** [reply]
sorry for interrupting your time, can you help me one more time. I'm looking for answers on google but haven't found what I'm looking for. is this wrong with my rabbitmq server? but  i have checked in console my rabbitmq server is running.
![image](https://user-images.githubusercontent.com/78366474/145977788-10a5fe3c-498e-44af-81ee-4c86c00cf3af.png)

**Reply by ShockwaveNN** [reply]
Judging by this port something wrong with RabbitMQ setup

Do you have it's installed?

I think Groups installation should download it

**Reply by mocakbarmaulana** [reply]
i have installed rabbitmq but this error appears. I also tried to install group without installing rabbitmq but still getting an error

**Reply by ShockwaveNN** [reply]
I think it's better to contact forum.onlyoffice.com - this is a place when people get support, because I don't think that we got any bug in our software, but you doing or skipping some crucial step during installation
